src/main/java/com/spotify/folsom/reconnect/ReconnectingClient.java

```diff
@@ -62,13 +63,14 @@ public ReconnectingClient(final BackoffFunction backoffFunction,
                             final boolean binary,
                             final Executor executor,
                             final long timeoutMillis,
-                            final Charset charset) {
+                            final Charset charset,
+                            final Metrics metrics) {
```

Comment:
If there is concern about anyone using the ReconnectingClient constructor directly instead of a MemcacheBuilder method, I could simply add an overload for the constructor so that the constructor with the old signature passes a NoOpMetrics